src/ShadowDOM/wrappers/Document.js

```diff
@@ -338,6 +338,13 @@
     'hasFeature',
   ]);
 
+  var originalCreateTreeWalker = document.createTreeWalker;
+  function createTreeWalker(root, whatToShow, filter, entityReferenceExpansion){
+    return originalCreateTreeWalker.call(document,unwrap(root), whatToShow, filter, entityReferenceExpansion);
+  }
+
+  document.createTreeWalker = createTreeWalker; // override
```

Comment:
oops, also the implementation will look something like:
```
    createTreeWalker: function() {
      return originalCreateTreeWalker.call(unwrap(this), unwrap(root), whatToShow, filter, entityReferenceExpansion);
    },
```

(note the "unwrap(this)" for the document itself)